engine/font/src/text_layout_skribidi.cpp

```diff
@@ -0,0 +1,252 @@
+}
+
+static bool LayoutText(LayoutContext* ctx,
+                        uint32_t* codepoints, uint32_t num_codepoints,
+                        TextLayoutSettings* settings,
+                         TextLayout* layout)
+{
+    HFontCollection font_collection = layout->m_FontCollection;
+
+    float line_width = settings->m_Width;
+    skb_layout_params_t params = {0};
+    params.font_collection    = FontCollectionGetSkribidiPtr(font_collection),
+    params.lang               = "en-us",                  // TODO: support setting
+    params.origin             = {0, 0},
+    params.layout_width       = line_width,
+    params.layout_height      = 1000000.0f,
+    params.base_direction     = SKB_DIRECTION_AUTO,
+    params.text_wrap          = (uint8_t)(settings->m_LineBreak ? SKB_WRAP_WORD : SKB_WRAP_NONE),
+    params.text_overflow      = SKB_OVERFLOW_NONE,
+    params.vertical_trim      = SKB_VERTICAL_TRIM_DEFAULT,
+    params.horizontal_align   = SKB_ALIGN_START,          // TODO: support the other way around (ask author for SKB_ALIGN_RIGHT/LEFT ?)
+    params.vertical_align     = SKB_ALIGN_START,          // TODO: support the other way around (ask author for SKB_ALIGN_RIGHT/LEFT ?)
+    params.baseline_align     = SKB_BASELINE_MIDDLE,
+    params.flags              = 0;
+
+    // TODO: Allo setting default as italic etc
+    const skb_attribute_t attributes[] = {
+        skb_attribute_make_font(SKB_FONT_FAMILY_DEFAULT, settings->m_Size, SKB_WEIGHT_NORMAL, SKB_STYLE_NORMAL, SKB_STRETCH_NORMAL),
+        skb_attribute_make_line_height(SKB_LINE_HEIGHT_METRICS_RELATIVE, settings->m_Leading),
+        skb_attribute_make_spacing(settings->m_Tracking, 0.0f)
+    };
+
+    // TODO: Support rich text
+
+    skb_text_run_utf32_t runs[] = {
+        { codepoints, (int32_t)num_codepoints, attributes, DM_ARRAY_SIZE(attributes) },
+    };
+
+    skb_layout_t* skblayout = skb_layout_create_from_runs_utf32(ctx->m_Alloc, &params, runs, DM_ARRAY_SIZE(runs));
+    ctx->m_Layout = skblayout;
+
+    const int32_t glyphs_count = skb_layout_get_glyphs_count(skblayout);
+    const skb_glyph_t* glyphs = skb_layout_get_glyphs(skblayout);
+    const skb_glyph_run_t* glyph_runs = skb_layout_get_glyph_runs(skblayout);
+    const skb_layout_params_t* layout_params = skb_layout_get_params(skblayout);
+    const skb_text_attributes_span_t* attrib_spans = skb_layout_get_attribute_spans(skblayout);
+    const skb_layout_line_t* layout_lines = skb_layout_get_lines(skblayout);
+    int32_t lines_count = skb_layout_get_lines_count(skblayout);
+
+    // alloc
+    uint32_t remaining_glyphs = layout->m_Glyphs.Remaining();
+    if (remaining_glyphs < glyphs_count)
+    {
+        layout->m_Glyphs.OffsetCapacity(glyphs_count - remaining_glyphs);
+    }
+
+    uint32_t remaining_lines = layout->m_Lines.Remaining();
+    if (remaining_lines < lines_count)
+    {
+        layout->m_Lines.OffsetCapacity(lines_count - remaining_lines);
+    }
+
+    const float edit_layout_y = 0;
```

Comment:
Here and below assign floats as `0.f` or `0.0f` to be consistent across the code